### PR TITLE
log failure in filter reload, use lowered permissions on initial filter-read

### PR DESCRIPTION
**Pull request by HeikoSchlittermann**
There is a long standing bug entry in the Debian packaged tinyproxy (Debian #756040).
I fixed this some years ago and I believe, the fix could be included in the upstream source.

Old behaviour:
Tinyproxy read the config and the filter file with the startup permissions (usually root).
After a HUP (reload), e.g. issued after regular logrotate, the child processes ran with lowered permissions and - depending on the perms of the filter file - they couldn't re-read the filter. And they didn't even iussed a notice about that failure.

New behaviour:
Read the filter definition *after* changing the uid of the tinyproxy processes, *and* write a log message there is a failure in reading the filter definition.

### CODE REVIEW

**Comment by rofl0r** [comment]
funny, was just about to remove the SIGHUP functionality at all... see: https://github.com/tinyproxy/tinyproxy/pull/218

File: src/child.c [hunk 1]
```diff
@@ -304,7 +304,8 @@ static void child_main (struct child_s *ptr)
                  * Make sure no error occurred...
                  */
                 if (connfd < 0) {
-                        log_message (LOG_ERR,
+                        if (errno != EAGAIN)    /* do not spam the logs */
```
Comment (rofl0r) [inline src/child.c]: under which circumstances do you get an EAGAIN here ?

**Comment by HeikoSchlittermann** [comment]
rofl0r <notifications@github.com> (Do 20 Dez 2018 03:14:04 CET):
> rofl0r commented on this pull request.
>
>
>
> > @@ -304,7 +304,8 @@ static void child_main (struct child_s *ptr)
>                   * Make sure no error occurred...
>                   */
>                  if (connfd < 0) {
> -                        log_message (LOG_ERR,
> +                        if (errno != EAGAIN)    /* do not spam the logs */
>
> under which circumstances do you get an EAGAIN here ?

When the child process gets the SIGHUP, the accept() gets interrupted
and returns EAGAIN, which is expected and should be fine.

    Best regards from Dresden/Germany
    Viele Grüße aus Dresden
    Heiko Schlittermann
--
 SCHLITTERMANN.de ---------------------------- internet & unix support -
 Heiko Schlittermann, Dipl.-Ing. (TU) - {fon,fax}: +49.351.802998{1,3} -
 gnupg encrypted messages are welcome --------------- key ID: F69376CE -
 ! key id 7CBF764A and 972EAC9F are revoked since 2015-01 ------------ -

**Comment by HeikoSchlittermann** [comment]
rofl0r <notifications@github.com> (Do 20 Dez 2018 03:11:59 CET):
> funny, was just about to remove the SIGHUP functionality at all... see: https://github.com/tinyproxy/tinyproxy/pull/218
>

Ok, you changed the model used for parallelism from pre-fork to pthread.
That's good, I believe, as long as there are no problems with shared
resources. (But I'm not very familiar with libpthread, so I can't tell.)

The change I introduced had the sole purpose to avoid delayed surprise
because the HUPped processes were not able to read the file (depending
on the permissions on the file - here root:root 640).

And somehow I touched the SIGHUP handling, as it seems :)

    Best regards from Dresden/Germany
    Viele Grüße aus Dresden
    Heiko Schlittermann
--
 SCHLITTERMANN.de ---------------------------- internet & unix support -
 Heiko Schlittermann, Dipl.-Ing. (TU) - {fon,fax}: +49.351.802998{1,3} -
 gnupg encrypted messages are welcome --------------- key ID: F69376CE -
 ! key id 7CBF764A and 972EAC9F are revoked since 2015-01 ------------ -

**Comment by rofl0r** [comment]
i think this PR is no longer needed; if i'm wrong on that please reopen.

**Pull request closed without merging**